README.md

```diff
@@ -1,8 +1,161 @@
+import React from 'react';
+import {connect} from 'react-algoliasearch-helper';
+
+export default connect()(
+  ({helper}) =>
+    <input
+      placeholder="Search here"
+      autoFocus
+      onChange={e => helper.setQuery(e.target.value).search()}
+    />
+);
+```
+
+Hits.js
+```js
+import React, {PropTypes} from 'react';
+import {connect} from 'react-algoliasearch-helper';
+
+const Results = ({results}) => {
+  if (!results) return <div/>;
+  return <div>{results.hits.map(hit => <div key={hit.objectID}>{hit.Name}</div>)}</div>;
+};
+
+Results.propTypes = {
+  results: PropTypes.object
+};
+
+export default connect(
+  state => ({results: state.searchResults})
+)(Results);
+```
+
+index.js
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import algoliasearch from 'algoliasearch';
+import algoliasearchHelper from 'algoliasearch-helper';
+import {Provider} from 'react-algoliasearch-helper';
+import SearchBox from './components/SearchBox.js';
+import Hits from './components/Hits.js';
+
+const client = algoliasearch('latency', 'ffc36feb6e9df06e1c3c4549b5af2b31');
+const helper = algoliasearchHelper(client, 'starbucks');
+
+const App = () =>
+  <Provider helper={helper}>
+    <div>
+      <SearchBox/>
+      <Hits/>
+    </div>
+  </Provider>;
+
+ReactDOM.render(<App/>, document.querySelector('#root'));
+```
+    "babel-plugin-add-module-exports": "^0.2.1",
+# API
+
+## `<Provider helper>`
+
+Makes the Algolia Search `helper` available to the `connect()` calls in the component hierarchy below. You can’t use connect() without wrapping your root component in <Provider>.
+
+### Props
+
+* `helper` ([algoliasearch-helper-js](https://github.com/algolia/algoliasearch-helper-js)):
+* `children` (ReactElement) The root of your component hierarchy.
+
+### Example
+
+```js
+ReactDOM.render(
+  <Provider helper={helper}>
+    <MyRootComponent />
+  </Provider>,
+  rootEl
+)
+```
+
+## `connect([mapStateToProps])(WrappedComponent)`
+
+Connects a React component to the helper.
+
+### Arguments
+
+* `[mapStateToProps(state, ownProps): stateProps]` (Function): if specified, the component will subscribe to helper events (`change`, `search`, `result`, `error`). Allowing you to compute props based on the helper state for your connected component. Its result must be a plain object, and it will be merged into the component’s props. If you omit it, the component will not be subscribed to the helper updates.
+
+### Remarks
+
+* It needs to be invoked two times. The first time with its arguments described above, and a second time, with the component: `connect([mapStateToProps])(WrappedComponent)`.
+* It does not modify the passed React component. It returns a new, connected component, that you should use instead.
+* The mapStateToProps function takes a single argument of the entire Redux store’s state and returns an object to be passed as props. It is often called a selector. Use reselect to efficiently compose selectors and compute derived data.
```

Comment:
*helper's state, and it's not exactly the helper state.

More like:

{ searchParameters: SearchParameters (helper state), searchResults: SearchResults, searching: bool, searchError: Error }

Also, I don't know if reselect is compatible with our library, but it might be.